### DISCUSSION TITLE
How to achieve an outerglow effect

### DISCUSSION THREAD

**Discussion by defunky** [ImageMagick/ImageMagick#5048] [Help] [answered]
Whats the best way to achieve an outer glow border around a transparent image similar to this:
![image](https://user-images.githubusercontent.com/17749501/163726967-8c9f4872-adbb-4660-8d06-b489602bc1fd.png)
Would it be to copy the base image, resize it and blur it? or is there a better method of achieving such an effect?

**Comment by fmw42** [comment]
Please post your input image and actual result. The image you posted is not transparent. Also please identify your IM version and platform/OS.

**Reply by defunky** [reply]
![square](https://user-images.githubusercontent.com/17749501/163727679-8c328e65-1d55-4c10-98b8-ad0d5e827837.png)

The attachment screwed up, It's just a simple square i'm trying to achieve glow effect around it. I don't have a result as I'm not sure how to achieve the effect. I'm using Magic.NET which is using IM 7; windows.

**Comment by fmw42** [comment, marked as answer]
There may be better ways to do it than this, but here is one way in command line. Sorry, I do not use Magick.Net.  

```
Read the image.
Extract the alpha channel and process it as follows:
Flood fill the center with white (so you have a white filled rectangle on black background)
Blur it and stretch appropriately to remove the blur effect inside the rectangle
Flood fill the center with black
Add that to the original alpha channel
Put the result into the alpha channel of the input replacing the old alpha channel
Save the result
```

Input:

![red_rect](https://user-images.githubusercontent.com/13458218/163730161-7e1e4e70-6515-4b06-a17a-422947763640.png)


```
convert red_rect.png \
\( -clone 0 -alpha extract \) \
\( -clone 1 -fill white -draw "color 110,175 floodfill" -alpha off \
-blur 0x7 -level 0x50% \
-fill black -draw "color 110,175 floodfill" -alpha off \) \
\( -clone 2,1 -compose plus -composite \) \
-delete 1,2 \
-alpha off -compose copy_opacity -composite \
red_rect3.png

```

![red_rect3](https://user-images.githubusercontent.com/13458218/163730699-1f7732f7-0d2c-483b-aff6-01f51b6a3d06.png)

Download the result to see the outer glow

**Reply by defunky** [reply]
Thank you so much, I'm learning the commands but could a similar effect be achieved by applying a shadow and with no offset values

**Reply by fmw42** [reply]
Yes, but it will still put a glow on the inside. So you have to flood fill the inside again afterwards.

```
convert red_rect.png \
\( -clone 0 -background white -shadow 80x7+0+0 \) +swap \
-background none -layers merge +repage \
\( -clone 0 -alpha extract -fuzz 99% -fill black -draw "color 110,175 floodfill" \) \
-alpha off -compose copy_opacity -composite \
red_rect4.png

```

![red_rect4](https://user-images.githubusercontent.com/13458218/163734528-514addfe-9720-4f70-afcb-a42a5dbee53a.png)
